lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -798,8 +798,14 @@ LoadedFile *SerializedModuleLoaderBase::loadAST(
       M.setABIName(Ctx.getIdentifier(loadedModuleFile->getModuleABIName()));
     if (loadedModuleFile->isConcurrencyChecked())
       M.setIsConcurrencyChecked();
-    if (!loadedModuleFile->getModulePackageName().empty())
+    if (!loadedModuleFile->getModulePackageName().empty()) {
+      if (loadedModuleFile->isBuiltFromInterface()) {
```

Comment:
Is this limited to modules of the same package? It looks like this will raise an error for swiftinterfaces in the SDK from an unrelated package.

Reply:
Package symbols are not printed in interface, so it should error regardless of whether it's in the same package

Reply:
makes sense; will update

Reply:
It looks good now, thanks!